x/programs/runtime/import_state.go

```diff
@@ -62,6 +64,20 @@ func NewStateAccessModule() *ImportModule {
 				defer cancel()
 				return callInfo.State.Insert(ctx, prependAccountToKey(callInfo.Account, parsedInput.Key), parsedInput.Value)
 			})},
+			"put_many": {FuelCost: writeManyCost, Function: FunctionNoOutput(func(callInfo *CallInfo, input []byte) error {
```

Comment:
We should think about how to support dynamically metered host function calls

Reply:
you can consume additional fuel within the host function via the callinfo.